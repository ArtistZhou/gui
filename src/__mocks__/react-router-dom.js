import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
