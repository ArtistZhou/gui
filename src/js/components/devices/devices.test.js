import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { createMount } from '@material-ui/core/test-utils';
import Devices from './devices';

const mockStore = configureStore([]);
const store = mockStore({
  devices: {
    byStatus: {
      accepted: { total: 0 },
      pending: { total: 0 }
    },
    selectedDeviceList: []
  }
});
store.dispatch = jest.fn();

it('renders correctly', () => {
  const tree = createMount()(
    <MemoryRouter>
      <Provider store={store}>
        <Devices />
      </Provider>
    </MemoryRouter>
  );
  expect(tree.html()).toMatchSnapshot();
});
