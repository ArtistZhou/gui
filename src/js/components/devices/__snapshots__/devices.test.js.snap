// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "marginTop": "-15px",
    }
  }
>
  <div
    className="MuiTabs-root-1"
  >
    <div
      className="MuiTabs-flexContainer-2"
    >
      <div
        className="MuiTabs-scroller-4 MuiTabs-fixed-5"
        onScroll={[Function]}
        role="tablist"
        style={
          Object {
            "marginBottom": 0,
          }
        }
      >
        <div
          className="MuiTabs-flexContainer-2"
        >
          <a
            aria-selected={true}
            className="MuiButtonBase-root-22 MuiTab-root-10 MuiTab-textColorInherit-12 MuiTab-selected-15"
            href="/devices"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex="0"
          >
            <span
              className="MuiTab-wrapper-18"
            >
              <span
                className="MuiTab-labelContainer-19"
              >
                <span
                  className="MuiTab-label-20"
                >
                  Device groups
                </span>
              </span>
            </span>
            <span
              className="MuiTouchRipple-root-155"
            />
          </a>
          <a
            aria-selected={false}
            className="MuiButtonBase-root-22 MuiTab-root-10 MuiTab-textColorInherit-12"
            href="/devices/pending"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex="0"
          >
            <span
              className="MuiTab-wrapper-18"
            >
              <span
                className="MuiTab-labelContainer-19"
              >
                <span
                  className="MuiTab-label-20"
                >
                  Pending
                </span>
              </span>
            </span>
            <span
              className="MuiTouchRipple-root-155"
            />
          </a>
          <a
            aria-selected={false}
            className="MuiButtonBase-root-22 MuiTab-root-10 MuiTab-textColorInherit-12"
            href="/devices/preauthorized"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex="0"
          >
            <span
              className="MuiTab-wrapper-18"
            >
              <span
                className="MuiTab-labelContainer-19"
              >
                <span
                  className="MuiTab-label-20"
                >
                  Preauthorized
                </span>
              </span>
            </span>
            <span
              className="MuiTouchRipple-root-155"
            />
          </a>
          <a
            aria-selected={false}
            className="MuiButtonBase-root-22 MuiTab-root-10 MuiTab-textColorInherit-12"
            href="/devices/rejected"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex="0"
          >
            <span
              className="MuiTab-wrapper-18"
            >
              <span
                className="MuiTab-labelContainer-19"
              >
                <span
                  className="MuiTab-label-20"
                >
                  Rejected
                </span>
              </span>
            </span>
            <span
              className="MuiTouchRipple-root-155"
            />
          </a>
        </div>
        <span
          className="MuiPrivateTabIndicator-root-25 MuiPrivateTabIndicator-colorSecondary-27 MuiTabs-indicator-9"
          style={
            Object {
              "left": 0,
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="tab-container"
  >
    <div
      className="leftFixed"
    >
      <div>
        <div
          className="muted margin-bottom-small"
        >
          Groups
        </div>
        <ul
          className="MuiList-root-28 MuiList-padding-29"
        >
          <div
            className="MuiButtonBase-root-22 MuiListItem-root-32 grouplist MuiListItem-default-35 MuiListItem-gutters-40 MuiListItem-button-41"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            style={
              Object {
                "backgroundColor": "#e7e7e7",
              }
            }
            tabIndex="0"
          >
            <div
              className="MuiListItemText-root-44"
            >
              <span
                className="MuiTypography-root-50 MuiTypography-subheading-57 MuiListItemText-primary-47"
              >
                <span>
                  All devices
                </span>
              </span>
            </div>
            <span
              className="MuiTouchRipple-root-155"
            />
          </div>
          <hr
            className="MuiDivider-root-86"
          />
          <div
            className="MuiButtonBase-root-22 MuiButtonBase-disabled-23 MuiListItem-root-32 grouplist MuiListItem-default-35 MuiListItem-gutters-40 MuiListItem-disabled-38 MuiListItem-button-41"
            onBlur={[Function]}
            onClick={null}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            style={
              Object {
                "color": "#d4e9e7",
              }
            }
            tabIndex="-1"
          >
            <div
              className="MuiListItemIcon-root-91"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-92"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
              </svg>
            </div>
            <div
              className="MuiListItemText-root-44"
            >
              <span
                className="MuiTypography-root-50 MuiTypography-subheading-57 MuiListItemText-primary-47"
              >
                Create a group
              </span>
            </div>
          </div>
        </ul>
      </div>
    </div>
    <div
      className="rightFluid"
      style={
        Object {
          "paddingTop": "0",
        }
      }
    >
      <div
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <button
          className="MuiButtonBase-root-22 MuiButton-root-101 MuiButton-text-103 MuiButton-textSecondary-105 MuiButton-flat-106 MuiButton-flatSecondary-108"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 100,
            }
          }
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiButton-label-102"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-92 buttonLabelIcon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
            Filters
          </span>
          <span
            className="MuiTouchRipple-root-155"
          />
        </button>
      </div>
      <div
        className="relative"
      >
        <div
          className="loaderContainer"
        >
          <div
            className="loader"
          >
            <span
              className="dot dot_1"
            />
            <span
              className="dot dot_2"
            />
            <span
              className="dot dot_3"
            />
            <span
              className="dot dot_4"
            />
          </div>
        </div>
        <div
          className="hidden"
        >
          <p>
            No devices found
          </p>
          <p>
            No devices have been authorized to connect to the Mender server yet.
          </p>
        </div>
        <div />
      </div>
    </div>
  </div>
</div>
`;
